plugins/Profiler/backend.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+'use strict';
+
+import type Bridge from '../../agent/Bridge';
+import type Agent from '../../agent/Agent';
+
+const emptyFunction = () => {};
+
+module.exports = (bridge: Bridge, agent: Agent, hook: Object) => {
+  const checkIfProfilingIsSupported = () => {
+    let profilingIsSupported = false;
+
+    // Feature detection for profiling mode.
+    // The presence of an "actualDuration" field signifies:
+    // 1) This is a new enough version of React
+    // 2) This is a profiling capable bundle (e.g. DEV or PROFILING)
```

Comment:
I forgot PROFILING is enabled in DEV too.

Is there any overhead in this implementation for DevTools running React in DEV mode while we're not recording?

Reply:
> Is there any overhead in this implementation for DevTools running React in DEV mode while we're not recording?

Yes and no. React will automatically opt roots into `ProfileMode` if DevTools is present. We have to do this in order to maintain accurate tree base times in the event that someone actually profiles.

But that behavior is in the React repo already, aside from this release. I don't think we have much new overhead in DevTools itself from this release _unless_ someone is actually profiling. (`ProfileCollector` doesn't gather data unless the user is recording.)